bigchaindb/backend/mongodb/connection.py

```diff
@@ -138,7 +139,7 @@ def _connect(self):
         except (pymongo.errors.ConnectionFailure,
                 pymongo.errors.OperationFailure) as exc:
             logger.info('Exception in _connect(): {}'.format(exc))
-            if "Authentication fail" in str(exc):
```

Comment:
This is just for follow-up:
We get `Authentication failed` with plain SCRAM auth, and `Could not find user ...` if there is no user in the database as specified in `database.login` parameter, and `auth failed` if the presented certificate has a different name than the one in the database, and the one specified in `database.login`.

@r-marques and I consulted and have decided that we will remove the `AuthenticationError` in BigchainDB and just log the specific error, along with raising it again.